screens/HotspotDetailScreen.tsx

```diff
@@ -234,6 +262,90 @@ export const HotspotDetailScreen = ({
 
   const address = formatHotspotAddress(hotspotDetails, false);
 
+  const deleteCat = async (cat: Cat) => {
+    if (
+      !hotspotDetails.sterilizedCats.includes(cat) &&
+      !hotspotDetails.unsterilizedCats.includes(cat)
+    ) {
+      cat.isSterilized ? setNewSterilizedCats([]) : setNewUnsterilizedCat([]);
+      return;
+    }
+    setIsInProgress(true);
+    const { success } = await deleteCatRequest(cat.id);
+
+    if (success) {
+      setIsInProgress(false);
+      SnackbarManager.success('Cat deleted!');
+
+      const unsterilizedCats = hotspotDetails.unsterilizedCats.filter(
+        (c: Cat) => c.id !== cat.id
+      );
+      const sterilizedCats = hotspotDetails.sterilizedCats.filter(
+        (c: Cat) => c.id !== cat.id
+      );
+
+      setHotspotDetails({
+        ...hotspotDetails,
+        unsterilizedCats: unsterilizedCats,
+        sterilizedCats: sterilizedCats,
+      });
+    } else {
+      setIsInProgress(false);
+      SnackbarManager.error(
+        'HotspotFormScreen - deleteCat func.',
+        'Cat not deleted'
+      );
+    }
+  };
+
+  const save = useHotspotSave({
+    hotspotToBeSaved: hotspotDetails,
+    shouldUpdate: true,
+    setIsInProgress,
+  });
+
+  const addNewCat = async (newCat: Cat) => {
+    if (hotspotDetails.id === undefined || hotspotDetails.id === '') {
+      const hotspot = await save();
```

Comment:
can the `id` also be `null` ?
Also, when is the hotspot id equal to the empty string?

Reply:
The `hotspotDetails.id` will be equal to a string if the first `useEffect` in `HotspotDetailScreen` receives an empty response from the server after calling `loadHotspotDetails`. 

Also, I'm not sure if the ID could ever have a `null` value, but it is sure that it will contain an `undefined` value if the request to the server fails.

Reply:
I decided to rewrite that function (the nested if statements in particular, but also the long condition).